website/src/controllers/api-v3/payments/amazon.js

```diff
@@ -0,0 +1,277 @@
+import amazonPayments from 'amazon-payments';
+import async from 'async';
+import cc from 'coupon-code';
+import mongoose from 'mongoose';
+import moment from 'moment';
+import nconf from 'nconf';
+import payments from './index';
+import shared from '../../../../common';
+import { model as User } from '../../models/user';
+
+const IS_PROD = nconf.get('NODE_ENV') === 'production';
+
+let api = {};
+
+let amzPayment = amazonPayments.connect({
+  environment: amazonPayments.Environment[IS_PROD ? 'Production' : 'Sandbox'],
+  sellerId: nconf.get('AMAZON_PAYMENTS:SELLER_ID'),
+  mwsAccessKey: nconf.get('AMAZON_PAYMENTS:MWS_KEY'),
+  mwsSecretKey: nconf.get('AMAZON_PAYMENTS:MWS_SECRET'),
+  clientId: nconf.get('AMAZON_PAYMENTS:CLIENT_ID'),
+});
+
+api.verifyAccessToken = function verifyAccessToken (req, res) {
+  if (!req.body || !req.body.access_token) {
```

Comment:
here the check for `req.body` can be removed, now it defaults to an empty object